[FIX] website_mass_mailing: editing the newsletter warning banner in …

jhk-odoo opened the pull request:
…the translation web editor

Steps to reproduce:
- install website, website_mass_mailing
- go to website, edit and add a popup window (you should see warning banner stating
that a newsletter popup is present on this window). Save.
- go to settings > languages > load a new language (fr_FR for example) and translate
the website
- go back to the website > select the new language > click the TRANSLATE button

Previous behavior:
the warning banner is not editable, also click the "Edit popup" button does nothing

Current behavior:
the banner is editable

opw-2280188

jhk-odoo commented:
wip

nle-odoo commented:
The warning popup is not very important to be editable if this is already translatable by Settings > Translated Terms.

If I remember correctly, you should be able to open the modal by pressing CTRL while clicking thanks to these code:

https://github.com/odoo/odoo/blob/157ffe55acd077d7c62007e35c1b145411734a2c/addons/website/static/src/js/editor/wysiwyg_multizone_translate.js#L248-L250

https://github.com/odoo/odoo/blob/157ffe55acd077d7c62007e35c1b145411734a2c/addons/website/static/src/js/editor/wysiwyg_multizone_translate.js#L268-L270

but testing in 13.0 I don't even enter this code, and in 11.0 I enter the popup still isn't shown

it would probably make sense that CTRL works, and the popup open

jhk-odoo commented:
i added an alert with an explanation for the "edit popup" link
kept the banner editable (needed for the click event to register)
switched the field to html_translate and moved the translation link slightly to the left

nle-odoo commented on addons/web/static/src/scss/form_view.scss:
```diff
@@ -611,6 +611,10 @@
             padding-right: 25px;
         }
     }
+    iframe+.o_field_translate {
```
I don't understand, there is no `+` in CSS, you can't target something from outside an `iframe`

jhk-odoo replied:
it selects an element right after another (changed it)

nle-odoo replied:
ah okay I did not know it worked without space so I tought it was a odd `+.` typo

nle-odoo commented on addons/website_mass_mailing/models/website_mass_mailing.py:
```diff
@@ -13,4 +14,4 @@ def _default_popup_content(self):
     mailing_list_id = fields.Many2one('mailing.list')
     website_id = fields.Many2one('website')
-    popup_content = fields.Html(string="Website Popup Content", default=_default_popup_content, translate=True, sanitize=False)
+    popup_content = fields.Html(string="Website Popup Content", default=_default_popup_content, translate=html_translate, sanitize=False)
```
I think you will kill any translation that exist for every database of odoo, you should not change it.

nle-odoo commented on addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js:
```diff
@@ -148,4 +149,18 @@ WysiwygMultizone.include({
+    start: function () {
+        this.$target.on('click.newsletter_popup_option', '.o_edit_popup', function (ev) {
+            // So that the snippet is not enabled again by the editor
+            alert(_t('Website popups can only be translated trough the Email Marketing app. You can access it by going to Email Marketing > Mailing Lists > List > ... | Edit > Website Popups.'));
```
```
Email Marketing > Mailing Lists > List > ... | Edit > Website Popups.
```

is a little too specific, these menu / helps are often renamed (and this probably would not be adapted here). I would just say something like:

> Website popups can only be translated trough mailing configuration in the Email Marketing app.

nle-odoo commented on addons/website_mass_mailing/views/snippets_templates.xml:
```diff
@@ -52,7 +52,7 @@
 <template id="s_newsletter_subscribe_popup" name="Newsletter Popup">
     <div class="o_newsletter_popup" data-list-id="0">
-        <div class="alert alert-warning css_non_editable_mode_hidden o_not_editable" role="alert">
+        <div class="alert alert-warning css_non_editable_mode_hidden o_editable" role="alert">
```
i'm not sure why this is needed, doesn't the alert show just by maintaining CTRL when clicking?

jhk-odoo replied:
it's not necessary to trigger the click, but it makes the banner editable

nle-odoo replied:
the newsletter is already editable

> ![2020-07-02-112034_1920x1080_scrot](https://user-images.githubusercontent.com/9977887/86340891-de893f00-bc55-11ea-98f7-29d9e58b2531.png)

If you mean editing this part:

![2020-07-02-112154_618x63_scrot](https://user-images.githubusercontent.com/9977887/86341002-01b3ee80-bc56-11ea-9205-6e9541094517.png)

it's part of the edition interface, there is no reason to edit it.

nle-odoo commented on addons/web/static/src/scss/form_view.scss:
```diff
@@ -611,6 +611,10 @@
             padding-right: 25px;
         }
     }
+    .oe_form_field_html .o_field_translate {
```
sorry, now it also impacts the inline editor that do not have the fullscreen button, so this was better before (but with space between plus)

![2020-07-02-130916_1444x557_scrot](https://user-images.githubusercontent.com/9977887/86351809-1fd51b00-bc65-11ea-8cba-6b18638dbb3f.png)

nle-odoo commented:
after last change, it's alright to merge

robodoo delegate+

jhk-odoo commented:
robodoo r+

nle-odoo commented:
I replaced `iframe` by `iframe.wysiwyg_iframe` in selector to be slightly more specific

robodoo r+
